src/components/dialogs/field-hook.js

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, {
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { elementExists } from '../../utils/rest-api';
+import { CircularProgress, InputAdornment, TextField } from '@material-ui/core';
+import CheckIcon from '@material-ui/icons/Check';
+import { useSelector } from 'react-redux';
+import { UploadCase } from './upload-case';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+
+const useStyles = makeStyles((theme) => ({
+    helperText: {
+        margin: 0,
+        marginTop: 4,
+    },
+}));
+
+export const useTextValue = ({
+    label,
+    id = label,
+    defaultValue = '',
+    adornment,
+    autoFocus = false,
+    triggerReset,
+    ...formProps
+}) => {
+    const [value, setValue] = useState(defaultValue);
+
+    const classes = useStyles();
+
+    const handleChangeValue = useCallback((event) => {
+        setValue(event.target.value);
+    }, []);
+
+    const field = useMemo(() => {
+        return (
+            <TextField
+                key={id}
+                size="small"
+                margin="dense"
+                id={id}
+                label={<FormattedMessage id={label} />}
+                value={value}
+                style={{ width: '100%' }}
+                onChange={handleChangeValue}
+                autoFocus={autoFocus}
```

Comment:
This seems strange. We should let the custom hook user set the autoFocus props to the desired field. If we have multiple `useTextValue` in a dialog for exemple, which one will be focused ?

Reply:
the one we put autofocus:true on the useTextValue :)